### PR TITLE
Fix https base tag generation.

### PR DESCRIPTION
**Pull request by GregoryRusakov**
Fixed https base tag generation.
Discussion is here:
https://github.com/joomla/joomla-cms/issues/4961#issuecomment-76674032

### CODE REVIEW

File: libraries/joomla/document/html/renderer/head.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ public function fetchHead($document)
 
 		if (!empty($base))
 		{
-			$buffer .= $tab . '<base href="' . $document->getBase() . '" />' . $lnEnd;
+			$buffer .= $tab . '<base href="' .Juri::base(). '" />' . $lnEnd;
```
Comment (n9iels) [inline libraries/joomla/document/html/renderer/head.php]: plese add spaces like `href="' . Juri::base() . '"`

**Comment by Hackwar** [comment]
Reading the code, it seems as if the base is set wrong somewhere. If you don't use getBase(), we can't influence this from for example a plugin. So why are you not fixing the place where the base in JDocument is set to a false value and instead replace the call with the static JURI::base()?

**Comment by nonumber** [comment]
I think the issue might be in ```libraries/cms/application/site.php```
Which has:
```
if ($router->getMode() == JROUTER_MODE_SEF)
{
	$document->setBase(htmlspecialchars(JUri::current()));
}
```
Shouldn't that be?
```
if ($router->getMode() == JROUTER_MODE_SEF)
{
	$document->setBase(htmlspecialchars(JUri::base()));
}
```

**Comment by Sulpher** [comment]
Peter provided solution, but it wasn't included in Joomla 3.4.3. Please add this patch in next Joomla version. Thank you.

**Comment by mbabker** [comment]
If someone opens a pull request with his suggested change and it gets tested then it can be included in a future release.  Until then, there isn't much that can be done.  http://magazine.joomla.org/issues/issue-sept-2015/item/2837-did-they-fix-it-yet offers some good insight on how you can help ensure issues that you are having with the core software get resolved :smile:

**Comment by Sulpher** [comment]
Michael, thanks for the link. Sure, it's important to test all patches.
Well, I've tested this patch on various sites with SSL and Varnish - it works. Since the status of this fix is marked as checked I thought it's already revised.

**Comment by mbabker** [comment]
The checks block just above the comment box here shows the status of our automated test suites, mainly to make sure that unit tests and codestyle checks continue to pass.  Unless a pull request has the `RTC` label or has already been merged, then it still needs human review and testing.

**Comment by Sulpher** [comment]
Ok, thanks for explanation. How many reviews are usually required to accept the patch?

**Comment by Bakual** [comment]
Usually two.

**Comment by b2z** [comment]
@GregoryRusakov can you update your PR to @nonumber solution? Seems that it is the right one.

**Comment by wojsmol** [comment]
@b2z I sent PR #7902 which implements solution suggested by @nonumber. @GregoryRusakov please test mentioned PR.

**Comment by zero-24** [comment]
Thanks @wojsmol so I'm closing here in favor of #7902 thanks

**Comment by wojsmol** [comment]
@Sulpher  please test PR.https://github.com/joomla/joomla-cms/pull/7902

**Pull request closed without merging**